CMakeLists.txt

```diff
@@ -15,31 +15,58 @@
 cmake_minimum_required(VERSION 3.18)
 project(manifold LANGUAGES CXX)
 
+# Use C++17
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
 set(CMAKE_VERBOSE_MAKEFILE ON)
 
-if(EMSCRIPTEN)
-  set(JSBIND_DEFAULT ON)
-else()
-  set(JSBIND_DEFAULT OFF)
-endif()
+# Define Manifold version
+set(MANIFOLD_VERSION_MAJOR 2)
+set(MANIFOLD_VERSION_MINOR 5)
+set(MANIFOLD_VERSION_PATCH 1)
+set(MANIFOLD_VERSION "${MANIFOLD_VERSION_MAJOR}.${MANIFOLD_VERSION_MINOR}.${MANIFOLD_VERSION_PATCH}")
 
-set(MANIFOLD_PAR "NONE" CACHE STRING "Parallel backend, either \"TBB\" or \"NONE\"")
-set(MANIFOLD_FLAGS "" CACHE STRING "Manifold compiler flags")
+
+# Primary user facing options
 option(MANIFOLD_CROSS_SECTION "Build CrossSection for 2D support" ON)
-option(MANIFOLD_EXPORT "Build mesh export (via assimp) utility library" OFF)
-option(MANIFOLD_TEST "Enable testing suite" ON)
-# fuzztest is a rather large dependency
-option(MANIFOLD_FUZZ "Enable fuzzing tests" OFF)
 option(MANIFOLD_DEBUG "Enable debug tracing/timing" OFF)
+option(MANIFOLD_DOWNLOADS "Allow Manifold build to download missing dependencies" ON) 
```

Comment:
This is the new compile option, right? So this is the main new feature of this PR, beyond the major simplification of our stucture and CMake.

Reply:
Yes - most of the time it will be a no-op, so it's not hugely significant, but it's handy for those projects that want to be sure their building of Manifold isn't downloading anything.